### ISSUE TITLE
Moped \ Project Staffing - Maintain project team history when removing project team members

### ISSUE THREAD

**Issue by zappfyllc** [cityofaustin/atd-data-tech#4400]

**Comment by amenity** [comment]
@zappfyllc - I think this might belong in a revision tracking/logging feature? I can't turn it up right now.

**Comment by zappfyllc** [comment]
@amenity I would move this up in the backlog to coincide with #4876 because I don't want us to forget about it. It is vital that we add in a mechanism for tracking project team changes because otherwise we will lose this data the second anyone edits the team in the newly added table in #4876.

**Comment by amenity** [comment]
@zappfyllc — I'm confused as to why change-tracking on project team is more important than change-tracking on other project information. 

In my mind, change-tracking of all project data — besides project extent — in the activity feed is a prerequisite for users managing real projects in Moped. Until then I'm not worrying too much about projects getting overwritten, updated, etc. since  all that we're working with will be for testing purposes. What am I missing? 🤔

**Comment by amenity** [comment]
Also, #4876 is explicitly for rendering a non-editable table. Last week I suggested we could make it the first piece of editable, but that would be specifically so we could use this issue, project team history, as our first use case for #4614. 

In the meantime I just want to see the team assigned during project creation — same as we can currently see Description, Current phase, etc. on the Summary page.

**Comment by zappfyllc** [comment]
> @zappfyllc — I'm confused as to why change-tracking on project team is more important than change-tracking on other project information.

I don't believe I said it's more important, so my apologies if that's the impression. The issue I'm highlighting here is this: all change-tracking may not be the same implementation (data stored in Postgres vs. data stored in a NoSQL database) and technically this user story was written for shipping the "Project Staffing" feature but obviously was dropped during the rush to create a demo. I 100% understand it wasn't important at the time, and it's partially my responsibility (since Leslie and I are the ones who originally drafted this feature), but I'd like to avoid that in the future because this was labeled a "Must Have" and preceded any sort of thorough requirements for the activity feed.

Additionally, we could've just handled that during moped_proj_personnel mutations to the database in the "New Project" workflow but because that workflow did not require the database in _close to_ real-time, it never was implemented that way, which was a gap on our end that's being identified in this issue.

> In my mind, change-tracking of all project data — besides project extent — in the activity feed is a prerequisite for users managing real projects in Moped. Until then I'm not worrying too much about projects getting overwritten, updated, etc. since all that we're working with will be for testing purposes. What am I missing? 🤔

I'm fine with the activity feed becoming MVP _if it's realistic to implement in a reasonable time_, but I'd just like to point out that this is a newer product decision. You can still have logging of changes (what we discussed during and after the roles/permissions implementation) without implementing a UI for users to see all changes, let alone revert them, and that is potentially a much more reasonable implementation in terms of delivery time/release. I also want to draw the distinction between project data (i.e. attributes a user edits directly in a project) and project-affiliated data (i.e. related objects with their own set of attributes that a user edits in a separate tab), which both still may need to be tracked, but may require different implementations (or at least a consideration of each implementation).

I totally understand the point about not worrying about projects being overwritten updated until it's shipped. I've only recently come around to the idea that there's just one larger release in May since that was also a recent decision, but point taken :).

**Comment by amenity** [comment]
@zappfyllc - @mddilley actually wrote up [this issue](https://github.com/cityofaustin/atd-data-tech/issues/5285) and, hot off the heels of implementing the "Project team" tables, is ready to work on it.

**Comment by amenity** [comment]
Too specific and small to keep open at this point.

**Comment by chiaberry** [comment]
Addressed while handling https://github.com/cityofaustin/atd-data-tech/issues/10810
